WarriorOfLeporia/WarriorOfLeporia/GameScene.swift

```diff
@@ -0,0 +1,225 @@
+//
+//  GameScene.swift
+//  WarriorOfLeporia
+//
+//  Created by Kirupa Nijastan on 2/28/18.
+//  Copyright © 2018 Kirupa Nijastan. All rights reserved.
+//
+
+import SpriteKit
+import GameplayKit
+
+struct PhysicsMasks {
+    static let None: UInt32 = 0 // 0
+    static let Player: UInt32 = 0b1 // 1
+    static let Enemy: UInt32 = 0b10 // 2
+    static let Bullet: UInt32 = 0b100 // 4
+    static let LevelBounds: UInt32 = 0b1000 // 8
+}
+
+
+class GameScene: SKScene, SKPhysicsContactDelegate {
+//    private var brabbit = SKSpriteNode()
+//    private var brabbitMoveFrames: [SKTexture] = []
+    var gameOver = false
+    var lastUpdateTime : TimeInterval = 0
+    
+    var leftButton: ButtonNode!
+    var rightButton: ButtonNode!
+    var jumpButton: ButtonNode!
+    
+    var shootButton: ButtonNode!
+    var spinButton: ButtonNode!
+    
+    var entityManager: EntityManager?
+    var player: PlayerEntity?
+    
+    override func didMove(to view: SKView) {
+     //   buildBRabbitTest()
+        // Added bounding box around level
+        let maxAspectRatio: CGFloat = 16.0/9.0
+        let maxAspectRatioHeight = size.width / maxAspectRatio
+        let playableMargin: CGFloat = (size.height - maxAspectRatioHeight)/2
+        let playableRect = CGRect(x: 0, y: playableMargin, width: size.width, height: size.height-playableMargin*2)
+        physicsBody = SKPhysicsBody(edgeLoopFrom: playableRect)
+        physicsWorld.contactDelegate = self
+        physicsBody!.categoryBitMask = PhysicsMasks.LevelBounds
+
+        // Add buttons
+        leftButton = ButtonNode(iconName: "grey-squareblock-tile1", text: "left", onButtonPress: leftPressed)
```

Comment:
All this button setup could be moved into another class that manages the interface.